core/tests/get_credit_card_test.rs

```diff
@@ -0,0 +1,54 @@
+#[cfg(test)]
+mod get_credit_card_test {
+    use lockbook_core::assert_matches;
+    use lockbook_core::service::api_service;
+    use lockbook_core::service::api_service::ApiError;
+    use lockbook_core::service::test_utils::{
+        generate_account, generate_monthly_account_tier, generate_root_metadata, test_credit_cards,
+    };
+    use lockbook_models::api::*;
+
+    #[test]
+    fn get_credit_card() {
+        let account = generate_account();
+        let (root, _) = generate_root_metadata(&account);
+
+        api_service::request(&account, NewAccountRequest::new(&account, &root)).unwrap();
+
+        api_service::request(
+            &account,
+            SwitchAccountTierRequest {
+                account_tier: generate_monthly_account_tier(
+                    test_credit_cards::GOOD,
+                    None,
+                    None,
+                    None,
+                ),
+            },
+        )
+        .unwrap();
+
+        let result = api_service::request(&account, GetCreditCardRequest {})
+            .unwrap()
+            .credit_card_last_4_digits;
+
+        assert_matches!(result.as_str(), test_credit_cards::GOOD_LAST_4);
+    }
+
+    #[test]
+    fn get_credit_card_does_not_a_stripe_customer() {
+        let account = generate_account();
+        let (root, _) = generate_root_metadata(&account);
+
+        api_service::request(&account, NewAccountRequest::new(&account, &root)).unwrap();
+
+        let result = api_service::request(&account, GetCreditCardRequest {});
+
+        assert_matches!(
+            result,
+            Err(ApiError::<GetCreditCardError>::Endpoint(
+                GetCreditCardError::NotAStripeCustomer
```

Comment:
Wouldn't this be `OldCardDoesNotExist`?

Reply:
What's the difference between the two?

Reply:
For a person to be a stripe customer, they must have at least gone through `switch_account_tier` and thus would have a `StripeUserInfo` in redis. In this situation, since `switch_account_tier` was never called, they don't have any credit cards, but more specifically they don't even have a record of them being a stripe customer.

Reply:
You're also kinda right though, there should not even be any `OldCardDoesNotExist` since it is never used.